### PR TITLE
Add flag to control reboot interval

### PR DESCRIPTION
**Pull request by hakonhall**
- Adds reboot-interval-in-days flag that are read on every 25m maintain().
 - Changes the reboot algorithm to evenly distribute reboots after 1x-2x reboot intervals after the last reboot. Reboots are now guaranteed to be scheduled before 2x reboot interval.
 - Removes the possibility of overriding durations from environment, which was almost never used.

### CODE REVIEW

File: flags/src/main/java/com/yahoo/vespa/flags/custom/NodeMaintainerDurations.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * {@code NodeRepositoryMaintenance} depends on various durations, e.g. the reboot interval. This class
+ * defines a serialization format for specifying such durations as a JSON object with keys matching
+ * the name (reboot_interval) and with the value being the duration in second as an integer.
```
Comment (freva) [inline flags/src/main/java/com/yahoo/vespa/flags/custom/NodeMaintainerDurations.java]: Let the names be the same as class names? That's what we use to enable/disable maintainers.

File: flags/src/main/java/com/yahoo/vespa/flags/custom/NodeMaintainerDurations.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+ *
+ * @author hakonhall
+ */
+public class NodeMaintainerDurations {
```
Comment (freva) [inline flags/src/main/java/com/yahoo/vespa/flags/custom/NodeMaintainerDurations.java]: Add `hashCode()` and `equals()`

File: flags/src/main/java/com/yahoo/vespa/flags/custom/NodeMaintainerDurations.java (hunk 1 above)
Comment (hakonhall) [reply flags/src/main/java/com/yahoo/vespa/flags/custom/NodeMaintainerDurations.java]: The current names are fairly descriptive, while attaching a duration to a Maintainer's subclass name may not be. Also, a Maintainer's interval may be different from the duration controlled here - for instance making smaller adjustments to fail_grace in main would typically not affect the NodeFailer's Maintainer interval.

**Review by freva — APPROVED** [review]
LGTM, but I think it would be better to get the flag value in `NodeRebooter`, thereby avoiding having to restart for this to take effect. The interval is in practice overwritten to 25 minutes anyway.

File: flags/src/main/java/com/yahoo/vespa/flags/custom/NodeMaintainerDurations.java (hunk 2 above)
Comment (hakonhall) [reply flags/src/main/java/com/yahoo/vespa/flags/custom/NodeMaintainerDurations.java]: NA

File: flags/src/main/java/com/yahoo/vespa/flags/custom/NodeMaintainerDurations.java (hunk 1 above)
Comment (hakonhall) [reply flags/src/main/java/com/yahoo/vespa/flags/custom/NodeMaintainerDurations.java]: NA

**Comment by hakonhall** [comment]
Done

**Review by freva — APPROVED** [review]
:1st_place_medal:

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/NodeRebooter.java [hunk 3]
```diff
@@ -49,13 +52,32 @@ protected void maintain() {
...
+        if (overdue.get().isNegative())
             return false;
-        else // schedule with a probability such that reboots of nodes are spread roughly over the reboot interval
-            return random.nextDouble() < (double) interval().getSeconds() / (double)rebootInterval.getSeconds();
+
+        // Use a probability such that each maintain() schedules the same number of reboots,
+        // as long as 0 <= overdue <= rebootInterval, with the last maintain() in that interval
+        // naturally scheduling the remaining with probability 1.
+
+        int configServers = 3;
```
Comment (mpolden) [inline node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/NodeRebooter.java]: Should be determined automatically by using `CuratorDatabaseClient#cluster`.

Comment (hakonhall) [reply node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/NodeRebooter.java]: Fixed in #11313

**Pull request merged**